Sources/Identity.swift

```diff
@@ -43,6 +44,8 @@ import Foundation
     }
 
     public func readyForEvent(_ event: Event) -> Bool {
+        if event.urlVariables, getSharedState(extensionName: IdentityConstants.SharedState.Configuration.SHARED_OWNER_NAME, event: event)?.status != .set { return false}
+
         return state.bootupIfReady(getSharedState: getSharedState(extensionName:event:),
                                    createXDMSharedState: createXDMSharedState(data:event:))
     }
```

Comment:
Let's reverse the order of these two checks, so that `bootupIfReady` is called before checking the Configuration shared state for a `getUrlVariables` request. The reason is, if the first event is `getUrlVariables` and a Configuration shared state is not set, then Identity aborts and waits for Configuration without setting an initial shared state. By calling `bootupIfReady` first, then even if Configuration has not set a state yet, Identity can still load/generate an ECID and set a shared state for other extensions.